packages/web/src/common/store/pages/signon/sagas.ts

```diff
@@ -798,108 +800,122 @@ function* signIn(action: ReturnType<typeof signOnActions.signIn>) {
-        yield* fork(audiusBackendInstance.updateCreator, {
-          ...signInResponse.user,
-          events: { referrer: signOn.referrer }
-        })
+    // TODO (PAY-3479): This is temporary until hedgehog is fully moved out of libs
+    yield* call(refreshHedgehogWallet)
+    const account: AccountUserMetadata | null = yield* call(
+      userApiFetchSaga.getUserAccount,
+      {
+        wallet: signInResponse.wallet
       }
+    )
 
-      yield* put(pushRoute(route || FEED_PAGE))
-
-      const trackEvent = make(Name.SIGN_IN_FINISH, { status: 'success' })
-      yield* put(trackEvent)
+    // Login succeeded but we found no account for the user (incomplete signup)
+    if (!account) {
+      yield* put(
+        signOnActions.openSignOn(false, Pages.PROFILE, {
+          accountAlreadyExisted: true
+        })
+      )
+      yield* put(toastActions.toast({ content: messages.incompleteAccount }))
+      return
+    }
 
-      yield* put(signOnActions.resetSignOn())
+    const { user } = account
 
-      const isNativeMobile = yield* getContext('isNativeMobile')
-      if (!isNativeMobile) {
-        // Reset the sign on in the background after page load as to relieve the UI loading
-        yield* delay(1000)
-      }
-      if (isNativeMobile) {
-        yield* put(requestPushNotificationPermissions())
-      } else {
-        setHasRequestedBrowserPermission()
-        yield* put(accountActions.showPushNotificationConfirmation())
-        if (signInResponse.user.handle === 'fbtest') {
-          yield put(pushRoute('/fb/share'))
-        }
-      }
-    } else if (
-      !signInResponse.error &&
-      signInResponse.user &&
-      !signInResponse.user.name
-    ) {
-      // Go to sign up flow because the account is incomplete
+    // Loging succeeded and we found a user, but it's missing name, likely
+    // due to incomplete signup
+    if (!user.name) {
       yield* put(
         signOnActions.openSignOn(false, Pages.PROFILE, {
           accountAlreadyExisted: true,
           handle: {
-            value: signInResponse.user.handle,
+            value: user.handle,
             status: 'disabled'
           }
         })
       )
       yield* put(toastActions.toast({ content: messages.incompleteAccount }))
 
-      const trackEvent = make(Name.SIGN_IN_WITH_INCOMPLETE_ACCOUNT, {
-        handle: signInResponse.handle
-      })
-      yield* put(trackEvent)
-    } else if (signInResponse.error && signInResponse.phase === 'FIND_USER') {
-      // Go to sign up flow because the account is incomplete
       yield* put(
-        signOnActions.openSignOn(false, Pages.PROFILE, {
-          accountAlreadyExisted: true
+        make(Name.SIGN_IN_WITH_INCOMPLETE_ACCOUNT, {
+          handle: user.handle
         })
       )
-      yield* put(toastActions.toast({ content: messages.incompleteAccount }))
-    } else {
+      return
+    }
+
+    // Now that we have verified the user is valid, run the account fetch flow,
```

Comment:
nice comments

Reply:
Heh, had to explain it to myself and decided to leave em in 🤷